### DISCUSSION TITLE
Content scaffolds

### DISCUSSION THREAD

**Discussion by saurlax** [withastro/roadmap#1078] [Proposal]
### Body

# Summary

I'm wondering if it's possible to add a command like `hexo new <title>` that automatically creates content through scaffolding templates.

# Background & Motivation

Obviously, some traditional CMS frameworks such as [Hexo](https://hexo.io/docs/writing.html) and [Hugo](https://gohugo.io/getting-started/quick-start/#add-content) provide scaffolding functionality. But after searching Astro's documentation and issues, I found that it doesn't seem to have this feature.

In addition, because Astro has a `content.config.ts` file for managing content. So I think it's entirely possible to add a scaffold function to `defineCollection()`, and use it to calculate the corresponding default value.

# Goals

Add an `astro new <collection> <path/to/file>` command to create new content.

This feature will be more powerful than typical scaffolds, as Astro's collection supports both content and data, making it also suitable for creating new data files.

# Example

Assuming `content.config.ts` is defined as follows:

```ts
import { defineCollection, z } from "astro:content";

export const collections = {
  blog: defineCollection({
    type: "content",
    schema: z.object({
      title: z.string(),
      date: z.date().default(() => new Date()),
      tags: z.array(z.string()),
    }),
    scaffold: (ctx: { title: string }) => ({
      title: ctx.title,
      date: new Date(),
      tags: [],
    }),
  }),
  friends: defineCollection({
    type: "data",
    schema: z.object({
      name: z.string(),
      url: z.string(),
      description: z.string().optional(),
    }),
    scaffold: (ctx) => ({
      name: "",
      url: "",
      description: "",
    }),
  }),
};
```

If you run `astro new blog hello-world.md`, it should automatically generate `src/content/blog/hello-world.md` containing the following content:

```markdown
---
# Perhaps some automated processing methods can be added to automatically convert it into `Hello World`
title: hello-world.md
date: 2024-12-20T04:19:24.080Z
tags: []
---

```

**Comment by delucis** [comment]
Noting that this is very similar to the proposal in #553.

I’ll close this one in favour of that earlier discussion, but thanks for the detailed write-up, I’m sure stuff could be shared between the two discussions.
